exotica/src/Scene.cpp

```diff
@@ -31,595 +31,14 @@
-                  tmp_shape, tmp_world->getObject(obj_id_[i]).get());
-          geo_world_.at(obj_id_[i]).push_back(g);
-          trans_world_.at(obj_id_[i])
-              .push_back(fcl::Transform3f(collision_detection::transform2fcl(
-                  tmp_world->getObject(obj_id_[i])->shape_poses_[j])));
-          fcl_world_.at(obj_id_[i])
-              .push_back(std::shared_ptr<fcl::CollisionObject>(
-                  new fcl::CollisionObject(
-                      g->collision_geometry_,
-                      collision_detection::transform2fcl(
-                          tmp_world->getObject(obj_id_[i])->shape_poses_[j]))));
-        }
-      }
-    }
-  }
-
-  void CollisionScene::updateCollisionRobot() {
-    for (auto& it : fcl_robot_) {
-      for (std::size_t i = 0; i < it.second.size(); ++i) {
-        collision_detection::CollisionGeometryData* cd =
-            static_cast<collision_detection::CollisionGeometryData*>(
-                it.second[i]->collisionGeometry()->getUserData());
-        it.second[i]->setTransform(collision_detection::transform2fcl(
-            ps_->getCurrentState().getCollisionBodyTransform(cd->ptr.link,
-                                                             cd->shape_index)));
-        it.second[i]->getTransform().transform(
-            it.second[i]->collisionGeometry()->aabb_center);
-      }
-    }
-  }
-
-  void CollisionScene::updateWorld(
-      const moveit_msgs::PlanningSceneWorldConstPtr& world) {
-    ps_->processPlanningSceneWorldMsg(*world);
-
-    if (compute_dist)
-      reinitializeCollisionWorld();
-  }
-
-  void CollisionScene::update(std::string joint, double value) {
-    try {
-      ps_->getCurrentStateNonConst().setVariablePosition(joint, value);
-    } catch (const std::exception& ex) {
-      throw_pretty("Exception while trying to update individual joint '"
-                   << joint << "': " << ex.what());
-    }
-    ps_->getCurrentStateNonConst().update(true);
-
-    // Update FCL robot link
-    if (compute_dist)
-      updateCollisionRobot();
-  }
-
-  void CollisionScene::update(Eigen::VectorXdRefConst x)
-  {
-    if (joint_index_.size() != x.rows())
-      throw_pretty("Size does not match, need vector size of "
-                   << joint_index_.size() << " but " << x.rows()
-                   << " is provided");
-
-    if (BaseType == BASE_TYPE::FIXED)
-    {
-      for (std::size_t i = 0; i < joint_index_.size(); i++)
-        ps_->getCurrentStateNonConst().setVariablePosition(joint_index_[i],
-            x(i));
-    }
-    else if (BaseType == BASE_TYPE::FLOATING)
-    {
-      ps_->getCurrentStateNonConst().setVariablePosition(
-          world_joint_ + "/trans_x", x(0));
-      ps_->getCurrentStateNonConst().setVariablePosition(
-          world_joint_ + "/trans_y", x(1));
-      ps_->getCurrentStateNonConst().setVariablePosition(
-          world_joint_ + "/trans_z", x(2));
-      KDL::Rotation rot = KDL::Rotation::EulerZYX(x(3), x(4), x(5));
```

Comment:
This is why never noticed the ZYX <> RPY bug in Kinematica in practice - Kinematica was never used for floating base and the MoveIt planning scene for sampling based planning was correct due to this.

Cf. #138